jellyfish/_room_api.py

```diff
@@ -93,42 +106,47 @@ def add_peer(self, room_id: str, options: PeerOptionsWebRTC) -> (str, Peer):
         """
 
         peer_type = "webrtc"
-        options = PeerOptions(options)
-        request = AddPeerRequest(type=peer_type, options=options)
+        json_body = AddPeerJsonBody(type=peer_type, options=options)
 
-        resp = self._room_api.add_peer(room_id, request)
+        resp = self._request(add_peer, room_id=room_id, json_body=json_body)
         return (resp.data.token, resp.data.peer)
 
     def delete_peer(self, room_id: str, peer_id: str) -> None:
         """Deletes peer"""
 
-        return self._room_api.delete_peer(room_id, peer_id)
+        return self._request(delete_peer, id=peer_id, room_id=room_id)
 
     def add_component(
         self, room_id: str, options: Union[ComponentOptionsHLS, ComponentOptionsRTSP]
-    ) -> Component:
+    ) -> Union[ComponentHLS, ComponentRTSP]:
         """Creates component in the room"""
 
         if isinstance(options, ComponentOptionsHLS):
             component_type = "hls"
         elif isinstance(options, ComponentOptionsRTSP):
             component_type = "rtsp"
         else:
-            raise ValueError(
-                "options must be either ComponentOptionsHLS or ComponentOptionsRTSP"
-            )
+            raise ValueError("options must be either ComponentOptionsHLS or ComponentOptionsRTSP")
 
-        options = ComponentOptions(options)
-        request = AddComponentRequest(type=component_type, options=options)
+        json_body = AddComponentJsonBody(type=component_type, options=options)
 
-        return self._room_api.add_component(room_id, request).data
+        return self._request(add_component, room_id=room_id, json_body=json_body).data
 
     def delete_component(self, room_id: str, component_id: str) -> None:
         """Deletes component"""
 
-        return self._room_api.delete_component(room_id, component_id)
+        return self._request(delete_component, id=component_id, room_id=room_id)
 
     def hls_subscribe(self, room_id: str, tracks: list):
         """subscribes hls component for tracks"""
 
-        return self._hls_api.subscribe_tracks(room_id, {"tracks": tracks})
+        subscription_config = SubscriptionConfig(tracks=tracks)
+
+        return self._request(subscribe_tracks, room_id=room_id, json_body=subscription_config)
+
+    def _request(self, method, **kwargs):
+        resp = method.sync(client=self._client, **kwargs)
+        if isinstance(resp, Error):
+            raise RuntimeError(resp.errors)
```

Comment:
@Rados13 Do you think it's ok to just throw a RuntimeError if a request should fail?
Before we had a couple more specific Exception types, such as `UnauthorizedException`, `NotFoundException`, `BadRequestException`